packages/dds/tree/src/simple-tree/leafNodeSchema.ts

```diff
@@ -80,6 +80,19 @@ function makeLeaf<Name extends string, const T extends ValueSchema>(
 	return new LeafNodeSchema(`com.fluidframework.leaf.${name}`, t);
 }
 
+export function isLeafNodeSchema(
+	schema: TreeNodeSchema,
+): schema is LeafNodeSchema<string, ValueSchema> {
+	return schema.kind === NodeKind.Leaf;
+}
+
+/**
+ * Returns the given schema as a {@link LeafNodeSchema} and throws an error if it isn't one.
+ */
+export function asLeafNodeSchema(schema: TreeNodeSchema): LeafNodeSchema<string, ValueSchema> {
+	return isLeafNodeSchema(schema) ? schema : fail("Schema is not a leaf schema");
+}
+
```

Comment:
these are very useful since someone can just do `instanceof LeafSchema` and get pretty much the same results.